src/preprocess/etl.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+
+import pathlib as pl
+import pandas as pd
+import time
+import numpy as np
+import yaml
+from sklearn.preprocessing import StandardScaler
+from sklearn.preprocessing import LabelEncoder
+
+
+__author__ = "Wei (Serena) Zou"
+__copyright__ = "Wei (Serena) Zou"
+__license__ = "mit"
+
+
+class Check:
+    """
+    Class to check path, data size and data type
+    """
+
+    def __init__(self, data_path):
+        self.data_path = data_path
+
+    def list_all_files(self) -> pd.DataFrame:
+        """
+        go through all the files in the data_analysis directory and build a DataFrame with the file name,
+        parent path and modified time.
+        Returns: a pandas df
+        """
+
+        all_files = []
+        for i in self.data_path.rglob('*.*'):
```

Comment:
why you need to list all files?

Reply:
this will be used as an input for function "check_path" (if the input file exits) and "convert" (find the most recent data if the input folder saves all the versions of historical data)